src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -1909,7 +1909,7 @@ private void CheckOptimisticIVTAccessGrants(DiagnosticBag bag)
 
                     if (conclusion == IVTConclusion.PublicKeyDoesntMatch)
                         bag.Add(ErrorCode.ERR_FriendRefNotEqualToThis, NoLocation.Singleton,
-                                                                      otherAssembly.Identity);
+                                                                      otherAssembly.Identity.ToString(), this.ContainingAssembly.Identity.ToString());
```

Comment:
I don't believe we should be calling `otherAssembly.Identity.ToString` here. That will cause it to be captured under the current UI culture. Diagnostics can have their messages realized under a different culture.

Reply:
`toString()` has been overridden to be culture-independent.

Reply:
Why use `ToString()` here rather than letting the message provider serialize the arguments?

Reply:
I usually prefer to be explicit on how things get serialized to strings -- Jared's comment reflects that this is usually a very deliberate decision and being explicit signals intent better, IMHO.